source/Halibut/Transport/IConnectionManager.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Diagnostics;
+using Halibut.Transport.Protocol;
+
+namespace Halibut.Transport
+{
+    public interface IConnectionManager : IDisposable
+    {
+        bool IsDisposed { get; }
+        IConnection AcquireConnection(ExchangeProtocolBuilder exchangeProtocolBuilder, IConnectionFactory connectionFactory, ServiceEndPoint serviceEndpoint, ILog log, CancellationToken cancellationToken);
+        Task<IConnection> AcquireConnectionAsync(ExchangeProtocolBuilder exchangeProtocolBuilder, IConnectionFactory connectionFactory, ServiceEndPoint serviceEndpoint, ILog log, CancellationToken cancellationToken);
+        void ReleaseConnection(ServiceEndPoint serviceEndpoint, IConnection connection);
```

Comment:
Can these sync methods be marked as `[obsolete]`

Reply:
Done!